content/Sessions/Sessions.q.md

```diff
@@ -0,0 +1,924 @@
+
+ - It creates the value `John` of the `$_SESSION["firstName"]` session variable.
+
+ - It assigns the value `John` to the `$_SESSION["firstName"]` session variable.
+
+ - It accesses the value `John` in the `$_SESSION["firstName"]` session variable.
+
+ - It removes the value `John` from the `$_SESSION["firstName"]` session variable.
+
+
+/// type=MS, id=3318ac8e-4bf0-4b76-a58b-c0439099a94e, answer=[2,5]
+
+In the `echo` statement on line 8, what does `session_id()` do?
+
+ - It sets the session ID to `test-session-123`.
+
+ - It returns the session ID `test-session-123`.
+
+ - It displays the session ID `test-session-123`.
+
+ - It removes the session ID `test-session-123`.
+
+ - It accesses the session ID `test-session-123`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+session_start();
+session_id("test-session-123");
+ 
+$_SESSION["firstName"] = "John";
+$_SESSION["lastName"] = "Miller";
+
+echo session_id() . ' ' . $_SESSION["firstName"] . ' ' . $_SESSION["lastName"];
+?>
+```
+/// type=MS, id=473eac0d-ace7-4577-a173-3b3cc9e50731, answer=[1,4]
+
+Add the statement `unset($_SESSION["firstName"]);` on line 7. Execute the program. What are its outputs?
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `test-session-123 John`.
+
+ - It prints `test-session-123 Miller`.
+
+ - It prints `test-session-123 John Miller`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+session_start();
+session_id("test-session-123");
+ 
+$_SESSION["firstName"] = "John";
+$_SESSION["lastName"] = "Miller";
+unset($_SESSION["firstName"]);
+echo session_id() . ' ' . $_SESSION["firstName"] . ' ' . $_SESSION["lastName"];
+?>
+```
+/// type=MS, id=b24c4bbe-5f5e-40db-8faf-2ef61dde4346, answer=[3,4,5]
+
+Why does adding the statement `unset($_SESSION["firstName"]);` produce an error?
+
+ - On line 7, the key `firstName` is enclosed in double quotes `""`. 
+
+ - On line 7, the `unset($_SESSION["firstName"]);` statement is invalid.
+
+ - The `$_SESSION["firstName"]` session variable is no longer accessible on line 8.
+
+ - The session data of the `$_SESSION["firstName"]` session variable does not exists.
+
+ - On line 7, the `unset($_SESSION["firstName"]);` statement removes the `$_SESSION["firstName"]` session variable.
+
+
+/// type=SS, id=8d649459-3d59-446a-ba4e-dc9dc4d5f03d, answer=[1]
+
+What does `unset($_SESSION["firstName"]);` do?
+
+ - It deletes the `$_SESSION["firstName"]` session variable.
+
+ - It defines the `$_SESSION["firstName"]` session variable.
+
+ - It returns the `$_SESSION["firstName"]` session variable.
+
+ - It displays the `$_SESSION["firstName"]` session variable.
+
+ - It accesses the `$_SESSION["firstName"]` session variable.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+session_start();
+
+if (!empty($_POST)) {
+    if (isset($_POST['username']) && isset($_POST['password'])) {
+        echo "Hello, " . $_POST['username'] . '.';
+    }
+}
+?>
```

Comment:
`if(!empty($_POST))` is not necessary, isset is already enough

Reply:
much better if we check `$_POST["submit']` than just `$_POST` for security purposes.